src/get-started/java-binding.md

```diff
@@ -0,0 +1,382 @@
+root hashes of its Merkelized collections. When using `AbstractService`, the
+root hash list must be defined in the schema class that implements
+`com.exonum.binding.service.Schema` interface; when implementing `Service`
+directly - in the service itself.
+
+### Service Transactions Description
+
+Exonum transactions allow you to perform modifying atomic operations with the
+storage. Transactions are executed sequentially, in the order determined by the
+consensus of the nodes in the network.
+
+#### Messages
+
+Transactions are transmitted by the service client as an Exonum message. The
+transaction data in the message can be serialized using an arbitrary algorithm
+supported by both the service client and the service itself.
+
+!!! note
+    In the current framework version each service must implement a method for
+    sending transactions of this service to the network. It is planned that this
+    mechanism will be standardized and moved to the core.
+    An example of a pseudo-code transaction handler is shown below. For an
+    implementation example, see
+    `com.exonum.binding.cryptocurrency.ApiController#submitTransaction`.
+
+!!! note "Example of a Pseudo-code Transaction Handler"
+    ```
+    def handle_tx(request):
+      var body = request.getBody()
+      try:
+        var message = txMessageFromBody(body)
+        var tx = service.convertTransaction(txMessage)
+        var hash = tx.hash()
+        node.submitTransaction(tx)
+        return ok(createTxResponse(hash))
+      catch IllegalArgumentException as e:
+        return badRequest("Invalid request: " + e)
+      catch VerificationError as e:
+        return badRequest("Invalid transaction: " + e)
+      catch Error as e:
+        log("Internal error:" + e)
+        return internalServerError()
+      ```
+
+#### Executable Transactions
+
+To correctly process a transaction, it must be transformed into an
+*executable transaction* (see `TransactionConverter` or
+`Service#convertTransaction` method) and transmitted to the framework using
+`Node#submitTransaction` method. The framework checks it, and if it is correct,
+broadcasts it to other nodes of the system. Other nodes, having received the
+transaction message, convert it into an executable transaction, also using the
+service transaction converter.
+
+An executable transaction is a class object that implements
+`com.exonum.binding.messages.Transaction` interface and defines transaction
+behavior and business logic. The interface implementations must define the
+transaction authentication rule (usually, the digital signature verification of
+the message) - `isValid` method; and the execution rule for the
+transaction - `execute` method.
+
+Ed25519 is a standard cryptographic system for digital signature of Exonum
+messages. We use `com.exonum.binding.crypto.CryptoFunctions#ed25519`
+method for obtaining the implementation of the function.
+
+The implementation of `Transaction#isValid` transaction authentication method
```

Comment:
Nit: `isValid` is not necessarily for authentication only; it can also verify internal constraints of the transaction (e.g., sender and receiver being separate entities in the transfer transaction in the cryptocurrency demo).

Reply:
Yes, but AFAIK we plan to move this functionality (message authentication) to the core framework, don't we? 

Also, there are currently three places where a user can check preconditions:
- a constructor
- `isValid`
- `execute`

Which is, IMO, a bad thing :) Therefore, I'd tell users to use it to authenticate the tx only until we move this method to core.